sourcegraph/sourcegraph
# (feat) add repository metadata to repo match search result items

Part of https://github.com/sourcegraph/sourcegraph/issues/43961.

This PR:
- Adds repository metadata (aka `keyValuePairs`) to search result items `commit|path|repo|symbol|path` if `repository-metadata` feature flag is enabled

## Test plan
- `sg start`
- Enable `repository-metadata` feature flag
- Add some key value pairs through API (currently no UI for adding) GQL mutation `addRepoKeyValuePair` API
- Search and check search results

## Screenshots
| Description | Screenshot |
| --: | :-- |
| Repo match | <img width="1728" alt="image" src="https://user-images.githubusercontent.com/6717049/231539259-6563b666-0609-4a3c-a5b1-c0dd43ba27dd.png"> |
| Commit match | <img width="1728" alt="image" src="https://user-images.githubusercontent.com/6717049/231539568-c4727d4d-8704-402c-a9b7-9dfcbfadba3b.png"> |
| Path match | <img width="1728" alt="image" src="https://user-images.githubusercontent.com/6717049/231539336-e6bf05fd-2aeb-42f9-b7dc-133d0868d61e.png"> |
| Symbol match | <img width="1728" alt="image" src="https://user-images.githubusercontent.com/6717049/231539388-a5b3a486-a4f8-4a8a-9b5d-ce172f86bf51.png"> |
| Content match | <img width="1728" alt="image" src="https://user-images.githubusercontent.com/6717049/231539517-22a8ec40-9b3e-4fbd-9d9a-af6955d92389.png"> |



## App preview:

- [Web](https://sg-web-erzhtor-add-repo-metadata-to.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
